purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -614,6 +614,58 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         } ?: retrievePurchaseInfo(identityManager.currentAppUserID, listener)
     }
 
+    /**
+     * This function will change the current appUserID.
+     * Typically this would be used after a log out to identify a new user without calling configure
+     * @param newAppUserID The new appUserID that should be linked to the currently user
+     * @param [listener] An optional listener to listen for successes or errors.
+     */
+    @JvmOverloads
+    internal fun logIn(
+        newAppUserID: String,
+        listener: LogInListener? = null
+    ) {
+        identityManager.currentAppUserID.takeUnless { it == newAppUserID }?.let {
+            identityManager.logIn(newAppUserID,
+                    onSuccess = { purchaserInfo, created ->
+                        dispatch {
+                            listener?.onReceived(purchaserInfo, created)
+                            sendUpdatedPurchaserInfoToDelegateIfChanged(purchaserInfo)
+                        }
+                        fetchAndCacheOfferings(newAppUserID, state.appInBackground)
+                    },
+                    onError = { error ->
+                        dispatch { listener?.onError(error) }
+                    })
+        }
+                ?: retrievePurchaseInfo(identityManager.currentAppUserID, receivePurchaserInfoListener(
+                        onSuccess = { purchaserInfo ->
+                            dispatch { listener?.onReceived(purchaserInfo, false) }
+                        },
+                        onError = { error ->
+                            dispatch { listener?.onError(error) }
+                        }
+                ))
+    }
+
+    /**
+     * Resets the Purchases client clearing the save appUserID. This will generate a random user
+     * id and save it in the cache.
+     * @param [listener] An optional listener to listen for successes or errors.
+     */
+    @JvmOverloads
+    internal fun logOut(listener: ReceivePurchaserInfoListener? = null) {
+        val error: PurchasesError? = identityManager.logOut()
```

Comment:
I noticed `reset` also does

`deviceCache.clearLatestAttributionData(identityManager.currentAppUserID)`

Do we need that here?

Reply:
I forgot to add a comment to request feedback for this, but it's something I'm torn on. I didn't do it on iOS, since `reset` doesn't do it there either. 
I wanted to maintain consistency with the `reset` method. 
However, it's not technically needed for a reset or logout. 
and, if after logging out you log in as the same user, attribution could be re-sent needlessly.

Reply:
note that `logOut` does call `clearLatestAttributionData`, but it happens in identityManager instead of Purchases, since it does seem to pertain to identity management. 
I just am not sure whether we should be doing it at all in a reset / logout scenario. @vegaro thoughts?

Reply:
I think the only reason we had it was to not let the SharedPreferences grow forever. Note that if we never remove it, it will always stay cached since it's dependent on the `identityManager.currentAppUserID`

Reply:
I'm not quite sure what I was thinking when I replied before 😂 but yeah, I actually am clearing attribution data, both on iOS and Android. It's just done in IdentityManager instead of purchases. 
And thinking about it further, I don't see a reason not to do it - it's just a cache, so the worst case here is that if you log in again with the same user and we've cleared the cache, we just send it an extra time. 
I think it's a good tradeoff for having cleaner shared preferences.